actions/lib/action.py

```diff
@@ -59,8 +56,38 @@ def __init__(self, config):
         if region:
             self.credentials['region'] = region
 
+        self.session = Session(aws_access_key_id=self.credentials['aws_access_key_id'],
+                               aws_secret_access_key=self.credentials['aws_secret_access_key'])
+
+        self.account_id = self.session.client('sts').get_caller_identity().get('Account')
+        self.cross_roles_arns = {
+            arn.split(':')[4]: arn for arn in self.config.get('action', {}).get('roles_arns', [])
+        }
+
         self.resultsets = ResultSets()
 
+    def assume_role(self, account_id):
+        ''' Assumes role and setup Boto3 session for the cross-account capability'''
+        if account_id == self.account_id:
+            return
+
+        try:
+            assumed_role = self.session.client('sts').assume_role(
+                RoleArn=self.cross_roles_arns[account_id],
+                RoleSessionName='StackStormEvents'
+            )
+            self.credentials.update({
+                'aws_access_key_id': assumed_role["Credentials"]["AccessKeyId"],
+                'aws_secret_access_key': assumed_role["Credentials"]["SecretAccessKey"],
+                'security_token': assumed_role["Credentials"]["SessionToken"]
+            })
+        except ClientError:
+            self.logger.error('Could not assume role on account with id: %s', account_id)
+            raise
+        except KeyError:
```

Comment:
How can you be certain that the KeyError here has to do with no finding cross region role ARN in the config file?  I see multiple access to dictionary using keys in the try-except scope above.  The KeyError can come from any of those references.

Reply:
I took into consideration that the "assume_role()" method of the boto3 module (https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sts.html#STS.Client.assume_role), if does not raise an error (ClientError), will definitely return a response as mentioned in the documentation. So, the only key that might be missing is the one from the cross_roles_arn dict.

Reply:
Resolved - there is now only one key access in the `try` block.